src/Data/List/Relation/Unary/All.agda

```diff
@@ -42,12 +42,14 @@ data All {A : Set a} (P : Pred A p) : Pred (List A) (a ⊔ p) where
 -- Operations on All
 
 module _ {P : Pred A p} where
+  uncons : ∀ {x xs} → All P (x ∷ xs) → P x × All P xs
+  uncons (px ∷ pxs) = px , pxs
 
   head : ∀ {x xs} → All P (x ∷ xs) → P x
-  head (px ∷ pxs) = px
+  head = Prod.proj₁ ∘ uncons
 
   tail : ∀ {x xs} → All P (x ∷ xs) → All P xs
-  tail (px ∷ pxs) = pxs
+  tail = Prod.proj₂ ∘ uncons
 
```

Comment:
Is there a reason why you are not adding `proj₁` and `proj₂` to the import list?
I don't think these introduce name clashes.

Reply:
Nope